src/Application/Actions/Donations/Update.php

```diff
@@ -550,4 +553,24 @@ private function handleGeneralStripeError(ApiErrorException $exception, Donation
 
         return null;
     }
+
+    private function ensurePaymentIntentIsGoodOrCancelAndThrow(Donation $donation, PaymentIntent $confirmedPaymentIntent, Request $request): void
```

Comment:
I wonder if this var name or a doc block could give a bit more info about what we mean by "good"? In particular, that it's specific to the status of a PI right after `confirm()`ing.

Reply:
Good question - I think since that isn't so clear it's probably better to move the if condition outside this function. Will refactor.